app/scheduling/directives/timeInput.js

```diff
@@ -14,10 +14,74 @@ schedulingApp.directive("timeInput", this.timeInput = function ($timeout) {
 			time: '=',
 			minuteStep: '@',
 			onChangeDelay: '@',
-			onChange: '&'
+			onChange: '&',
+			floor: '=',
+			ceiling: '='
 		},
 		link: function (scope, element, attrs) {
 			var linkMinuteHour = (attrs.linkMinuteHour === 'true');
+			scope.canDecrementHours = true;
+			scope.canDecrementMinutes = true;
+			scope.canIncrementHours = true;
+			scope.canIncrementMinutes = true;
+
+			scope.updateUI = function() {
+				scope.canDecrementHours = true;
+				scope.canDecrementMinutes = true;
+				scope.canIncrementHours = true;
+				scope.canIncrementMinutes = true;
+
+				var selectedTime = moment(scope.time, "HH:mm:ss");
+				selectedMinutes = selectedTime.minutes();
+				selectedHours = selectedTime.hours();
+
+				if (selectedMinutes == "55") {
+					scope.canIncrementMinutes = false;
+				}
+
+				if (selectedMinutes == "00") {
+					scope.canDecrementMinutes = false;
+				}
+
+				if (scope.floor) {
+					var floorTime = moment(scope.floor, "HH:mm:ss");
+					floorTime = floorTime.add(5, "minutes");
+
+					floorMinutes = floorTime.minutes();
+					floorHours = floorTime.hours();
+
+
+					if (floorHours == selectedHours && floorMinutes == selectedMinutes) {
+						scope.canDecrementMinutes = false;
```

Comment:
Just a thought on this: it might be easier if each UI button (start hour up, start hour down, start minute up, start minute down, etc.) called a method called validTime() which took the current start time, end time, and the desired change and simply returned true/false if it made sense. You can pass them as times and/or use MomentJS and just do a subtraction between two dates. That would get rid of the string checking.

I'm not sure if that's better or worse than what's here, just curious if you've thought about that approach.